x-pack/plugins/cases/public/components/system_actions/cases/cases_params.test.tsx

```diff
@@ -0,0 +1,248 @@
+    timeWindow: '6w',
+    reopenClosedCases: false,
+    groupingBy: [],
+  },
+};
+
+const connector: ActionConnector = {
+  secrets: {},
+  config: {},
+  id: 'test',
+  actionTypeId: '.test',
+  name: 'Test',
+  isPreconfigured: false,
+  isDeprecated: false,
+  isSystemAction: false as const,
+};
+const editAction = jest.fn();
+const defaultProps = {
+  actionConnector: connector,
+  actionParams,
+  editAction,
+  errors: { 'subActionParams.timeWindow.size': [] },
+  index: 0,
+  producerId: 'test',
+};
+
+describe('CasesParamsFields renders', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    useApplicationMock.mockReturnValue({ appId: 'management' });
+    useAlertDataViewsMock.mockReturnValue({
+      loading: false,
+      dataViews: [
+        {
+          title: '.alerts-test',
+          fields: [
+            {
+              name: 'host.ip',
+              type: 'ip',
+              esTypes: ['keyword'],
+            },
+            {
+              name: 'host.geo.location',
+              type: 'geo_point',
+            },
+          ],
+        },
+      ],
+    });
+  });
+
+  it('all params fields are rendered', async () => {
+    render(<CasesParamsFields {...defaultProps} />);
+
+    expect(await screen.findByTestId('group-by-alert-field-combobox')).toBeInTheDocument();
+    expect(await screen.findByTestId('time-window-size-input')).toBeInTheDocument();
+    expect(await screen.findByTestId('time-window-unit-select')).toBeInTheDocument();
+    expect(await screen.findByTestId('reopen-case')).toBeInTheDocument();
+  });
+
+  it('renders loading state of grouping by fields correctly', async () => {
+    useAlertDataViewsMock.mockReturnValue({ loading: true });
+    render(<CasesParamsFields {...defaultProps} />);
+
+    expect(await screen.findByRole('progressbar')).toBeInTheDocument();
+  });
+
+  it('disables dropdown when loading grouping by fields', async () => {
+    useAlertDataViewsMock.mockReturnValue({ loading: true });
+    render(<CasesParamsFields {...defaultProps} />);
+
+    expect(await screen.findByRole('progressbar')).toBeInTheDocument();
+    expect(await screen.findByTestId('comboBoxSearchInput')).toBeDisabled();
+  });
+
+  it('when subAction undefined, sets to default', () => {
+    const newProps = {
+      ...defaultProps,
+      actionParams: {},
+    };
+    render(<CasesParamsFields {...newProps} />);
+    expect(editAction.mock.calls[0][1]).toEqual('run');
+  });
+
+  it('when subActionParams undefined, sets to default', () => {
+    const newProps = {
+      ...defaultProps,
+      actionParams: {
+        subAction: 'run',
+      },
+    };
+    render(<CasesParamsFields {...newProps} />);
+    expect(editAction.mock.calls[0][1]).toEqual({
+      timeWindow: '7d',
+      reopenClosedCases: false,
+      groupingBy: [],
+      owner: 'cases',
+    });
+  });
+
+  it('sets owner default if appId not matched', () => {
+    useApplicationMock.mockReturnValue({ appId: 'testAppId' });
```

Comment:
What do you think of adding a test wherein the first render the `appId` is `undefined`, then we rerender the component and on the second render we return a valid appId that much to security solution? You can chain calls to `useApplicationMock` like:

```
useApplicationMock.mockReturnValueOnce({ appId: undefined }).mockReturnValueOnce({ appId: 'testAppId' });
```

Reply:
Added in https://github.com/elastic/kibana/pull/179409/commits/c67258238cf90cecb683a14ebd6998e22aa175c6